apps/meteor/ee/app/livechat-enterprise/server/lib/OnHoldHelper.ts

```diff
@@ -0,0 +1,105 @@
+import { Meteor } from 'meteor/meteor';
+import type { IMessage, IOmnichannelRoom, IUser } from '@rocket.chat/core-typings';
+import { isOmnichannelRoom } from '@rocket.chat/core-typings';
+import { LivechatRooms, Subscriptions, Messages, LivechatInquiry } from '@rocket.chat/models';
+
+import { RoutingManager } from '../../../../../app/livechat/server/lib/RoutingManager';
+import { callbacks } from '../../../../../lib/callbacks';
+import { ohLogger as logger } from './logger';
+
+class OnHoldHelperClass {
+	async placeRoomOnHold(room: IOmnichannelRoom, comment: string, onHoldBy: IUser) {
+		logger.debug(`Attempting to place room ${room._id} on hold by user ${onHoldBy?._id}`);
+
+		const { _id: roomId } = room;
+
+		if (!room || !isOmnichannelRoom(room)) {
+			throw new Error('error-invalid-room');
+		}
+		if (!room.open) {
+			throw new Error('error-room-already-closed');
+		}
+		if (room.onHold) {
+			throw new Error('error-room-is-already-onHold');
+		}
+
+		const onHoldByUser: IMessage['u'] = {
+			_id: onHoldBy._id,
+			username: onHoldBy.username || '',
+			name: onHoldBy.name || '',
+		};
+
+		await Promise.all([
+			LivechatRooms.setOnHoldByRoomId(roomId),
+			Subscriptions.setOnHoldByRoomId(roomId),
+			Messages.createOnHoldHistoryWithRoomIdMessageAndUser(roomId, onHoldByUser, comment, 'on-hold'),
+		]);
+
+		Meteor.defer(() => {
```

Comment:
Do we need the meteor.defer? can't we move this to the end of func?

Reply:
Apparently it works without defer, so I'm gonna remove it since it's from Meteor and we're moving this to a service which is known to sometimes cause issues with meteor